setup.py

```diff
@@ -157,7 +157,7 @@ def build_extension(self, ext: CMakeExtension):
         )
 
         # Ensure that catalyst shared object is copied to the build directory for pip editable install
-        if backend in ("lightning_kokkos"):
+        if platform.system() == "Linux" and backend in ("lightning_kokkos"):
             source = os.path.join(f"{extdir}", f"lib{backend}_catalyst.so")
             destination = os.path.join(os.getcwd(), "build")
             shutil.copy(source, destination)
```

Comment:
Do we have any problem with building shared libraries for Mac? If not, we could check for Mac and move the `.dylib` file. Maybe some similar for Windows? (I forgot if this was supposed to build for Windows.)

Reply:
I mean, Catalyst will not work with Windows, but it is prepared to look for shared libraries in Mac systems.

Reply:
It should build shared libraries for Mac too, I can check if moving `.dylib` instead of `.so` works

Reply:
On Mac it works with copying dylib, I've made a new commit now

Reply:
Great! Thank you so much for checking that!